### PR TITLE
Add proper map serialization support in serix

### PR DESCRIPTION
**Pull request by muXxer**
There were functions in serix to serialize maps, but without the use of own `Serializable` implementations, it was impossible to use the maps properly. 

This PR changes that.

### CODE REVIEW

File: serializer/serix/serix.go [hunk 1]
```diff
@@ -886,6 +771,55 @@ func (api *API) parseStructType(structType reflect.Type) ([]structField, error)
...
+}
+
+func parseLengthPrefixType(prefixTypeRaw string) (LengthPrefixType, error) {
+	switch prefixTypeRaw {
+	case "byte", "uint8":
+		return LengthPrefixTypeAsByte, nil
+	case "uint16":
+		return LengthPrefixTypeAsUint16, nil
+	case "uint32":
+		return LengthPrefixTypeAsUint32, nil
```
Comment (alexsporn) [inline serializer/serix/serix.go:803]: uint64 was added to the serializer, so maybe add it here too?

**Review by alexsporn — APPROVED** [review]

File: serializer/serix/map_encode.go [hunk 2]
```diff
@@ -259,13 +276,41 @@ func (api *API) mapEncodeSlice(ctx context.Context, value reflect.Value, valueTy
...
+		return "", nil, ierrors.Wrapf(err, "failed to encode map key of type %s", key.Type())
+	}
+
+	v, err := api.mapEncode(ctx, val, valueTypeSettings, opts)
+	if err != nil {
+		return "", nil, ierrors.Wrapf(err, "failed to encode map element of type %s", val.Type())
+	}
+
+	//nolint:forcetypeassert // map keys are always strings
+	return k.(string), v, nil
```
Comment (PhilippGackstatter) [inline serializer/serix/map_encode.go:298]: Is the "map keys can only be strings" requirement documented somewhere?

**Review by PhilippGackstatter — APPROVED** [review]

File: serializer/serix/map_encode.go (hunk 2 above)
Comment (muXxer) [reply serializer/serix/map_encode.go:298]: No.... that was not even my code... it was always this way :rekt:

**Comment by muXxer** [comment]
Superseeded by #603 because there we remove the map rules again.

**Pull request closed without merging**